### PR TITLE
fix: add more check in cluster node parsing

### PR DESCRIPTION
**Pull request by jonathanc-n**
references #2401 

Added non confusing message when multiple nodes are added without a newline separator

### CODE REVIEW

**Review by PragmaTwice — COMMENTED** [review]
Could you add a test case for this patch?

**Comment by jonathanc-n** [comment]
Test case is added + the node checking is more flexible

**Comment by jonathanc-n** [comment]
@PragmaTwice Are you able to see what exactly is causing the fail on the error, I don't seem to be able to see the reason for it.

**Comment by git-hulk** [comment]
@jonathanc-n #2401 is caused by missing a newline between nodes, and omitting the newline at the end should be allowed.

**Comment by jonathanc-n** [comment]
@git-hulk Alright, it should be fully working now.

**Comment by git-hulk** [comment]
@jonathanc-n You can enhance this inside [parseClusterNodes](https://github.com/apache/kvrocks/blob/2a0c57ac7fc4499a210bc452884025a0010cb470/src/cluster/cluster.cc#L712)

**Comment by jonathanc-n** [comment]
@PragmaTwice @git-hulk This should be good I believe, I moved the changes to cluster.cc.

**Comment by git-hulk** [comment]
@jonathanc-n It would be better to check if the field is node id in line 778? https://github.com/apache/kvrocks/blob/2a0c57ac7fc4499a210bc452884025a0010cb470/src/cluster/cluster.cc#L778

The current solution is too tricky from my perspective.

**Comment by jonathanc-n** [comment]
@git-hulk I simplified as much as I thought I could for the issue to not persist. I believe it should be fine.

**Comment by jonathanc-n** [comment]
@git-hulk @PragmaTwice Pretty sure this should be good, it passed the workflow. Unless you see some changes that should be made.

File: src/cluster/cluster.cc [hunk 1]
```diff
@@ -716,6 +716,13 @@ Status Cluster::parseClusterNodes(const std::string &nodes_str, ClusterNodes *no
     return {Status::ClusterInvalidInfo, errInvalidClusterNodeInfo};
   }
 
+  for (const auto &entry : nodes_info) {
+    if (entry.find(' ') == std::string::npos) {
+      return {Status::ClusterInvalidInfo,
+              "Invalid nodes definition: Each node entry must contain fields separated by spaces."};
+    }
+  }
+
```
Comment (git-hulk) [inline src/cluster/cluster.cc]:
```suggestion
```

It has been checked at Line 730.

File: src/cluster/cluster.cc [hunk 2]
```diff
@@ -776,6 +783,13 @@ Status Cluster::parseClusterNodes(const std::string &nodes_str, ClusterNodes *no
     for (unsigned i = 5; i < fields.size(); i++) {
       std::vector<std::string> ranges = util::Split(fields[i], "-");
       if (ranges.size() == 1) {
+        std::regex node_id_regex(R"(\b[a-fA-F0-9]{40}\b)");
```
Comment (git-hulk) [inline src/cluster/cluster.cc]: It should be good to only check if `fields[i]` is a node id?

**Comment by jonathanc-n** [comment]
@git-hulk Alright, should be fine now, sorry about that.

**Comment by git-hulk** [comment]
@jonathanc-n No worry, thanks for your great contribution.

**Review by git-hulk — APPROVED** [review]

**Pull request merged**